server/src/sim.rs

```diff
@@ -149,7 +250,11 @@ impl Sim {
                     postcard::to_stdvec(pos.local.as_ref()).unwrap(),
                 ));
             }
-            if let Some(ch) = entity.get::<&Character>() {
+            if let Some(ch) = entity.get::<&Character>().or_else(|| {
+                entity
+                    .get::<&InactiveCharacter>()
+                    .map(|ich| hecs::Ref::map(ich, |ich| &ich.0)) // Extract Ref<Character> from Ref<InactiveCharacter>
+            }) {
```

Comment:
When you try to get a semantically equivalent field from two different possible component types, that's a strong signal that you should normalize that field into its own freestanding component (perhaps `struct Characterid(String)`) instead. Perhaps in a follow-up commit?

Reply:
I may have to work on this in a follow-up PR. There are too many controversial open questions related to this. There are several properties that could correspond to players:
- their unique ID
- a friendly name they might give themselves
- whether they are currently logged in or not
- their location in the world
- their `CharacterState` (consisting of physics-related info and client-provided orientation)
- the status of their inventory
- their current `CharacterInput`

Part of the reason I've been uncertain about the use of the `InactiveCharacter` struct is that it is conceptually unclear to me why the presence of `InactiveCharacter` vs `Character` is a good source of truth for whether a player is logged in or not. Why aren't any other structs, like `Inventory`, wrapped? I could imagine an "inventory stealer" entity that takes items from anything that has an inventory and a position, and, with the current implementation, there would be nothing stopping it from taking items from logged-out players.

It almost seems like we want to use `InactivePosition` instead of `InactiveCharacter`, since position is more fundamental, and it will pretty much stop all system logic from interacting with such entities. Also, using `InactivePosition` would seal the fate of having to get a semantically equivalent field from two different possible component types. Also, even `InactivePosition` doesn't cover everything because any system that just lists active players will need to check for the presence of a component in an unnatural way.

We could add an inactive counterpart to a whole bunch of components, but messing with more components than we need to seems like it will be more annoying long-term.

When more features are added in the future, the answers to these questions will hopefully become clearer, but for now, I feel like doing "something that works" is probably sufficient. I'm not convinced that this one example is enough to show that `Character` should be split.